### ISSUE TITLE
Script throws unexpected error when using optional enum method parameter with default value

### ISSUE THREAD

**Issue by daveaglick** [dotnet/roslyn#2272]
To reproduce:
```
ScriptOptions scriptOptions = new ScriptOptions()
	.AddNamespaces(
		"System",
		"System.IO")
	.AddReferences(
		Assembly.GetAssembly(typeof(object)),  // System
		Assembly.GetAssembly(typeof(Path)));  // System.IO
Script script = CSharpScript.Create(@"		
	public class MyClass
	{
		public void MyMethod(System.IO.SearchOption so = System.IO.SearchOption.AllDirectories)
		{
		}
	}
");
script.Run();
```

Result:
```
InvalidCastException: Invalid cast from 'System.Int32' to 'System.IO.SearchOption'.
```

Suspect it has to do with this line in `ReflectionEmitter.cs`:
```
if (paramType.IsEnum)
{
    paramType = paramType.UnderlyingSystemType;
}
```

which should probably be:
```
if (paramType.IsEnum)
{
    paramType = Enum.GetUnderlyingType(paramType);
}
```

PR is incoming...

**Comment by tmat** [comment]
We should stop using Reflection.Emit in scripting scenarios.

**Comment by KevinH-MS** [comment]
Fixed with 8e61348

**Comment by MattGertz** [comment]
Needs verification.
